addon/src/teardown-context.ts

```diff
@@ -30,7 +32,14 @@ export default function teardownContext(
     .then(() => {
       _cleanupOnerror(context);
 
-      destroy(context);
```

Comment:
I think the only potential impact of this change would be that if consumers are calling `registerDestructor` on the context for their own cleanup, then that would break and they would need to call it on `context.owner` instead.